src/test/java/dk/cngroup/lentils/service/StatisticsServiceIntegrationTest.java

```diff
@@ -0,0 +1,292 @@
+package dk.cngroup.lentils.service;
+
+import dk.cngroup.lentils.LentilsApplication;
+import dk.cngroup.lentils.entity.Cypher;
+import dk.cngroup.lentils.entity.CypherStatus;
+import dk.cngroup.lentils.entity.Hint;
+import dk.cngroup.lentils.entity.Team;
+import dk.cngroup.lentils.repository.TeamRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = LentilsApplication.class)
+@Transactional
+public class StatisticsServiceIntegrationTest {
+
+    @Autowired
+    private TeamRepository teamRepository;
+
+    @Autowired
+    private StatusService statusService;
+
+    @Autowired
+    private CypherService cypherService;
+
+    @Autowired
+    private GameLogicService gameLogicService;
+
+    @Autowired
+    private ObjectGenerator generator;
+
+    @Autowired
+    private StatisticsService statisticsService;
+
+    @Autowired
+    private HintTakenService hintTakenService;
+
+    @Autowired
+    private HintService hintService;
+
+    @Test
+    public void getNumbersOfTakenHintsOfCyphersTest() {
+        Cypher cypher1 = cypherService.save(generator.generateValidCypher());
+        Cypher cypher2 = cypherService.save(generator.generateValidCypher());
+        Team team1 = teamRepository.save(generator.generateTeamWithNameAndPin("team1", "1111"));
+        Team team2 = teamRepository.save(generator.generateTeamWithNameAndPin("team2", "2222"));
+        Team team3 = teamRepository.save(generator.generateTeamWithNameAndPin("team3", "3333"));
+
+        Hint hint1 = hintService.save(new Hint("hint1", 3, cypher1));
+        Hint hint2 = hintService.save(new Hint("hint2", 6, cypher1));
+        Hint hint3 = hintService.save(new Hint("hint2", 3, cypher2));
+        Hint hint4 = hintService.save(new Hint("hint3", 6, cypher2));
+
+        hintTakenService.takeHint(team1, hint1);
+        hintTakenService.takeHint(team1, hint2);
+        hintTakenService.takeHint(team2, hint1);
+        hintTakenService.takeHint(team3, hint2);
+
+        hintTakenService.takeHint(team1, hint3);
+        hintTakenService.takeHint(team1, hint4);
+        hintTakenService.takeHint(team2, hint3);
+        hintTakenService.takeHint(team2, hint4);
+        hintTakenService.takeHint(team3, hint3);
+
+        Map<Cypher, Integer> takenHintsOfAllCyphers = statisticsService.getNumberOfHintsTakenOfCyphers();
+
+        assertEquals((Integer) 4, takenHintsOfAllCyphers.get(cypher1));
```

Comment:
Dotaz asi spíše pro @mjanousek a @crlikcngroup: přijde mi více readable, pokud by tyhle asserty byly následovně: assertEquals(4, takenHintsOfAllCyphers.get(cypher1).intValue());
Má to nějaký vliv i na výkon, pokud se nebude int castovat na Integer, ale vyberem jen intValue z Integeru?

Reply:
Nejde napsat jen 4 misto (Integer)4?
Tzn zbavit se explicitniho pretypovani?

Reply:
Vykon asi nepoznas v tomhle pripade.

Reply:
Odpovim si sam, cast tam byt musi :) dobry.

Reply:
![image](https://user-images.githubusercontent.com/43091754/62217688-5e09f080-b3ab-11e9-8b9a-102d08b54385.png)

Reply:
nechat to tak, nebo to upravit na ....intValue() ?

Reply:
Ta varianta od @JurcaPavel se mi zda asi citelnejsi ale nevidim asi problem ani v te originalni. Pokud byste chteli se toho zbavit uplne, pak si muzete napsat funkci treba "assertIntegerEquals" nebo nejakou podobnou, ktera bude brat patricne parametry a ten cast se udela v ni. Ale pro ty dva radky mi to prijde asi zbytecne.. Necham na @daweedek2 co se mu zda nejlepsi ;)

Reply:
pridal jsem intValue() na konec tech assertu